README.md

```diff
@@ -1,29 +1,123 @@
 
 To verify that you're logged into correct project, you can run `oc whoami` and `oc project`.
 
-Running the tests amounts to standard `mvn clean verify`.
-This will use a specific Quarkus version, which can be modified by setting the `quarkus.platform.version` property.
-Alternatively, you can use `-Dquarkus-core-only` to run the test suite against Quarkus `999-SNAPSHOT`.
-In that case, make sure you have built Quarkus locally prior to running the tests.
-Moreover, all the modules are grouped by types: `http`, `security`, `messaging`... Hence we can run a concrete set of scenarios 
-just by selecting the right profile: `-P http-modules` or `-P security-modules`. Take into account that if you are selecting 
-another profile as OpenShift (`mvn clean verify -Dopenshift`), you need to explicitly to add `-Dall-modules` to include all 
-the modules in all the profiles.
+By default [Quarkus-test-framework](https://github.com/quarkus-qe/quarkus-test-framework) use [ephemeral namespaces](https://github.com/quarkus-qe/quarkus-test-framework/wiki/6.-Openshift#disable-ephemeral-namespaces) in order to be deterministic and don't populate you environment with random objects. After each scenarios all the objects that are required, will be removed.
+
+Practical Example:
+
+User: `I would like to run http-minimum module.` 
+
+```shell
+mvn clean verify -Dall-modules -Dopenshift -pl http/http-minimum 
+```
+
+### Openshift / Native
+
+Please read [Openshift](#Openshift) section first and login into OCP.
+
+When we are running a [native compilation](https://quarkus.io/guides/building-native-image) the flow is the same as the regular way, the only difference is that we need to compile our application first with GraalVM/Mandrel in order to generate the binary application. To do that we will add the flag `native` to our maven command.   
+
+Practical Example:
+
+User: `I would like to run http-minimum module.`
+
+```shell
+mvn clean verify -Dall-modules -Dnative -Dopenshift -pl http/http-minimum 
+```
+
+Quarkus test framework will reuse the Native binary generated by Maven to run the test, except if the scenario provides a build property, then it will generate a new Native binary.
+
+More info about how to generate native images could be found in Quarkus [building-native-image](https://quarkus.io/guides/building-native-image) website, and customize your native image generation by adding some [configuration flags](https://quarkus.io/guides/building-native-image#configuration-reference) to your maven command.
+
+For example, 
+
+User: `I want to use a custom GraalVM Docker image and 5GB of memory`
+
+```shell
+mvn clean verify -Dall-modules -Dnative -Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-native-image:21.3-java11 -Dquarkus.native.native-image-xmx=5g -Dopenshift -pl http/http-minimum 
+```
+
+Or what about use the GraalVM that I have installed in my local environment instead of use a Docker image with GraalVM
```

Comment:
These two use cases: local graal && customized docker graal should be more emphasized.
Two subchapters of ### Openshift / Native can be introduced for that. And move the content to them